sapmrequestparser/parser.go

```diff
@@ -0,0 +1,82 @@
+package sapmrequestparser
+
+import (
+	"bytes"
+	"compress/gzip"
+	"errors"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"sync"
+
+	"github.com/golang/protobuf/proto"
+	splunksapm "github.com/signalfx/sapm-proto/gen"
+)
+
+const (
+	// TraceEndpointV2 is the endpoint used for SAPM v2 traces.  The SAPM protocol started with v2.  There is no v1.
+	TraceEndpointV2   = "/v2/trace"
```

Comment:
I would vote to keep them and even make public/exported so that whoever implements SAPM client or server can use these pre-defined consts. These are part of protocol specification. Maybe only name them slightly differently, e.g.
```go
ContentTypeHeaderName = "Content-Type"
ContentTypeHeaderValue = "application/x-protobuf"
etc...
```
What do you think?

Reply:
SGTM

Reply:
This way SAPM implementations will refer to them as `sapmrequestparser.ContentTypeHeaderName`, probably not ideal but should be fine (something like `sapmprotocol.ContentTypeHeaderName` would look nicer, maybe put them in a different package?). Anyway up to you @codesmith14 , these are nits.